jwst/resample/resample.py

```diff
@@ -409,13 +383,15 @@ def resample_many_to_one(self, input_models):
             max_ctx_id=len(input_models),
             disable_ctx=False,
         )
-        self._init_variance_arrays()
+
         self._init_exptime_counters()
 
         log.info("Resampling science and variance data")
 
-        leading_group_idx = [v[0] for v in input_models.group_indices.values()]
+        invalid_var = 4 * [False]
+        var_list = ["var_rnoise", "var_flat", "var_poisson", "err"]
```

Comment:
Question about this part - is it scientifically equivalent to resample the rnoise/flat/poisson components separately and add them later?  If so, I would think that would be more computationally efficient than directly resampling the error array as well as the 3 components.

Reply:
_If_ `err` is _strictly_ `=sqrt(poisson+flat+rnoise)` and nothing else, then I guess they should be equivalent. However, I don't know that.

If this is true then why do we even save ERR arrays in the data models as they only take up memory...

Reply:
It is exactly that, up to the resampling stage.  I think it's just a user convenience to have the error estimate in a separate array, rather than having to add the components in quadrature themselves.

Reply:
What should be done (w.r.t. `err`) if one of the variance arrays is "missing" (i.e., is `None`)?

Reply:
It should be treated as an array full of zero.  This is only likely to happen for the flat variance - if the flat_field step is skipped, var_flat is None, and in that case, the variance component for flat fielding really is zero.

Reply:
That's not what current code would do: https://github.com/spacetelescope/jwst/blob/e86066cd882ccf2fd7cd6129582c21bbb784e176/jwst/resample/resample.py#L529-L531 Treating a missing input variance array as an array of zeros of the same shape as input data would result in output (resampled) variance array of zeros which is currently not the case. I am OK with changing how this behaves but I would probably prefer this to be done in a future/separate PR from this one as I am trying to reproduce the old behavior in order to minimize the amount of failing tests. It is just easier to track/explain changes.